### ISSUE TITLE
openvpn.conf: line 1 server-bridge option has missing or extra parameters.

### ISSUE THREAD

**Issue by janvda** [Salvoxia/docker-openvpn-tap#2]
When I run command:

```
docker run -v ovpn-data:/etc/openvpn -d --network host  --cap-add=NET_ADMIN salvoxia/openvpn-tap:0.4
```

I see the following in my openvpn container log file:

```
Identity added: /Users/jan/.ssh/id_rsa (jan@Jans-MBP.lan)
2024-01-28 18:42:00 TUN/TAP device tap0 opened
2024-01-28 18:42:00 Persist state set to: ON
Setting up bridge
setting tap device to promiscous mode
setting eth device to promiscous mode
setting IP, subnet and broadcast address for bridge
setting mac address for bridge
checking if default gateway needs to be added for bridge
setting up IPTABLES
Checking IPv6 Forwarding
Sysctl error for default forwarding, please run docker with '--sysctl net.ipv6.conf.default.forwarding=1'
Sysctl error for all forwarding, please run docker with '--sysctl net.ipv6.conf.all.forwarding=1'
Running 'openvpn --config /etc/openvpn/openvpn.conf --client-config-dir /etc/openvpn/ccd --crl-verify /etc/openvpn/crl.pem '
Options error: Unrecognized option or missing or extra parameter(s) in /etc/openvpn/openvpn.conf:1: server-bridge (2.6.8)
Use --help for more information.
 ```

This is the first line of my  `openvpn.conf` file:

```
server-bridge 192.168.255.0 255.255.255.0 255.255.255.0 192.168.2.200 192.168.2.220
```

So it has too many parameters (5 instead of 4)
I am also not sure if the first parameter is correct as I have nowhere specified IP address `192.168.255.0`

FYI this is what I have specified:

```
docker run -v ovpn-data:/etc/openvpn --rm  my-openvpn-dev-ovpn-server ovpn_genconfig \
        -u udp://$VPN_SERVER:$VPN_PORT \
        -t \
        -B \
        --bridge-name 'br0' \
        --bridge-eth-if 'enp3s0' \
        --bridge-eth-ip '192.168.2.11' \
        --bridge-eth-subnet '255.255.255.0' \
        --bridge-eth-broadcast '192.168.2.255' \
        --bridge-eth-gateway '192.168.2.1' \
        --bridge-eth-mac '9XXXXXXXXXXX3' \
        --bridge-dhcp-start '192.168.2.200' \
        --bridge-dhcp-end '192.168.2.220'
```

**Comment by Salvoxia** [comment]
`server-bridge 192.168.255.0 255.255.255.0` is derived from the value of the `-u` argument. So maybe the `$VPN_SERVER` you specified causes an issue with that logic. What did you set `$VPN_SERVER` to?

**Comment by janvda** [comment]
`$VPN_SERVER=niaXXXXX.be`

`XXXXX` must be replaced by proper string.


Note also that the number of arguments is also wrong.

https://openvpn.net/community-resources/ethernet-bridging/

<img width="858" alt="image" src="https://github.com/Salvoxia/docker-openvpn-tap/assets/17238398/9f175361-0217-4496-8cb2-b1803be51e08">

**Comment by janvda** [comment]
I see following logic in the script:

```
SERVER_DIRECTIVE="server $(getroute $OVPN_SERVER)"
if [ "${OVPN_USE_BRIDGE}" == "1" ]; then
    SERVER_DIRECTIVE="server-bridge $(getroute $OVPN_SERVER) $OVPN_BR_ETH_SUBNET $OVPN_BR_DHCP_START $OVPN_BR_DHCP_END"
fi
```

So it is using `$OVPN_SERVER`
but I am not providing this information as input parameter.
So it will use what is specified in line 179 =  `OVPN_SERVER=192.168.255.0/24``

So based on:

```
# Convert 1.2.3.4/24 -> 255.255.255.0
cidr2mask()
{
    local i
    local subnetmask=""
    local cidr=${1#*/}
    local full_octets=$(($cidr/8))
    local partial_octet=$(($cidr%8))

    for ((i=0;i<4;i+=1)); do
        if [ $i -lt $full_octets ]; then
            subnetmask+=255
        elif [ $i -eq $full_octets ]; then
            subnetmask+=$((256 - 2**(8-$partial_octet)))
        else
            subnetmask+=0
        fi
        [ $i -lt 3 ] && subnetmask+=.
    done
    echo $subnetmask
}

# Used often enough to justify a function
getroute() {
    echo ${1%/*} $(cidr2mask $1)
}
```


It is normal that 

```
$(getroute $OVPN_SERVER)
```

returns 

```
192.168.255.0 255.255.255.0
```


I think that following command at line 425 is not correct:

```
SERVER_DIRECTIVE="server-bridge $(getroute $OVPN_SERVER) $OVPN_BR_ETH_SUBNET $OVPN_BR_DHCP_START $OVPN_BR_DHCP_END"
```

It should be:

```
SERVER_DIRECTIVE="server-bridge $OVPN_BR_ETH_IP $OVPN_BR_ETH_SUBNET $OVPN_BR_DHCP_START $OVPN_BR_DHCP_END"
```

**Comment by Salvoxia** [comment]
> It should be:
> `SERVER_DIRECTIVE="server-bridge $OVPN_BR_ETH_IP $OVPN_BR_ETH_SUBNET $OVPN_BR_DHCP_START $OVPN_BR_DHCP_END"`


Thank you for your analysis, you came to the same conclusion as I did ;)
I'll be able to fix this and build a new release by this evening I guess.

**Comment by janvda** [comment]
Thanks for the swift response.  There is no hurry.

I am just wondering how come that you have not noticed this issue as it is showstopper for me.

**Comment by Salvoxia** [comment]
Haha, yeah , I asked that myself. I guess while developing these changes for this image there was a lot of testing and manual config editing involved. Also, you are the first to bring this up in about a year that this image has existed, so I guess interest has not been too overwhelming ;)

**Comment by Salvoxia** [comment]
Fixed with [0.5](https://github.com/Salvoxia/docker-openvpn-tap/releases/tag/0.5)
